pkg/analyzer/analyzers/dockerhub/dockerhub.go

```diff
@@ -0,0 +1,194 @@
+//go:generate generate_permissions permissions.yaml permissions.go elevenlabs
+package dockerhub
+
+import (
+	"errors"
+	"os"
+
+	"github.com/fatih/color"
+	"github.com/jedib0t/go-pretty/table"
+
+	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/analyzers"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/analyzer/config"
+	"github.com/trufflesecurity/trufflehog/v3/pkg/context"
+)
+
+var _ analyzers.Analyzer = (*Analyzer)(nil)
+
+type Analyzer struct {
+	Cfg *config.Config
+}
+
+// SecretInfo hold the information about the token generated from username and pat
+type SecretInfo struct {
+	User         User
+	Valid        bool
+	Reference    string
+	Permissions  []string
+	Repositories []Repository
+	ExpiresIn    string
+	Misc         map[string]string
+}
+
+// User hold the information about user to whom the personal access token belongs
+type User struct {
+	ID       string
+	Username string
+	Email    string
+}
+
+// Repository hold information about each repository the user can access
+type Repository struct {
+	ID        string
+	Name      string
+	Type      string
+	IsPrivate bool
+	StarCount int
+	PullCount int
+}
+
+func (a Analyzer) Type() analyzers.AnalyzerType {
+	return analyzers.AnalyzerTypeDockerHub
+}
+
+func (a Analyzer) Analyze(_ context.Context, credInfo map[string]string) (*analyzers.AnalyzerResult, error) {
+	username, exist := credInfo["username"]
+	if !exist {
+		return nil, errors.New("username not found in the credentials info")
+	}
+
+	pat, exist := credInfo["pat"]
+	if !exist {
+		return nil, errors.New("personal access token(PAT) not found in the credentials info")
+	}
+
+	info, err := AnalyzePermissions(a.Cfg, username, pat)
+	if err != nil {
+		return nil, err
+	}
+
+	return secretInfoToAnalyzerResult(info), nil
+}
+
+// AnalyzePermissions will collect all the scopes assigned to token along with resource it can access
+func AnalyzePermissions(cfg *config.Config, username, pat string) (*SecretInfo, error) {
+	// create the http client
+	client := analyzers.NewAnalyzeClientUnrestricted(cfg) // /user/login is a non-safe request
+
+	var secretInfo = &SecretInfo{}
```

Comment:
optional: This is purely a personal preference, so feel free to disregard. When passing a pointer to functions responsible for populating a struct, I find it clearer to declare the variable without `&`. This forces the reviewer to recognize that the pointer is being passed explicitly, indicating that we're modifying a pre-existing struct. Again, this is entirely optional.

Ex:
```
var secretInfo SecretInfo
...
if err := decodeTokenToSecretInfo(token, &secretInfo); err != nil {
  ...
}

if err := fetchRepositories(client, username, token, &secretInfo); err != nil {
  ...
}
```